### PR TITLE
add ability for specify version range for commandline commands

### PR DESCRIPTION
**Pull request by deviator**
no breaking changes expected

### CODE REVIEW

**Comment by deviator** [comment]
ping @Geod24
I don't sure what this change will be useful for many people, but I think it can be helpful when dub integrates with other systems (IDE or buildsystems).

**Comment by deviator** [comment]
ping @thewilsonator, this change like https://github.com/dlang/dub/pull/2057, but for other commands

**Comment by Geod24** [comment]
Sounds useful, will review now.

File: changelog/add-ability-for-specify-version-range-for-commandline-commands.dd [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Improvment of version specification for commandline commands
+
+Before this release dub was can get only strict version in commands like
+`describe`, `generate`, `fetch` and other. By now version can be specified
+same as dependency version:
+
+    dub fetch foo@'>0.2.0'
+    dub describe foo@'>=0.3.0 <1.0.0'
```
Comment (Geod24) [inline changelog/add-ability-for-specify-version-range-for-commandline-commands.dd:8]:
```suggestion
All commands now accept a version specification

Before this release dub could only get an exact version for some commands
(`describe`, `generate`, `fetch`, etc...). All commands now accept a version specification,
such as can be found in `dub.json` / `dub.sdl`:

    dub fetch 'foo@>0.2.0'
    dub describe foo@'>=0.3.0 <1.0.0'
    
Note that commands such as `describe` will still not fetch from the network.
```

Is this an accurate description ?

**Review by Geod24 — APPROVED** [review]
What are the affected / unaffected commands ?

File: changelog/add-ability-for-specify-version-range-for-commandline-commands.dd (hunk 1 above)
Comment (deviator) [reply changelog/add-ability-for-specify-version-range-for-commandline-commands.dd:8]: Only commands are child for `PackageBuildCommand` are affected: `generate`, `build`, `run`, `test`, `lint`, `describe` (`dustmite` have no version spec). No effects for `clean-caches`, `clean`, `add`, `upgrade`, `fetch`, `install`, and other (pr for `list` https://github.com/dlang/dub/pull/2057).

**Comment by deviator** [comment]
CyberShadow/DAutoTest failed:
```
posix.mak:807: recipe for target 'test' failed
make: *** [test] Error 1
Test failed: object.Exception@ae/sys/d/manager.d(877): Command ["timeout", "1800", "make", "-f", "posix.mak", "DIFFABLE=1", "test"] failed with status 2
```
but in log before
```
posix.mak:42: ===== DEPRECATION NOTICE ===== 
posix.mak:43: ===== DEPRECATION: posix.mak is deprecated. Please use src/build.d instead.
posix.mak:44: ============================== 
```
may it depends?

**Comment by Geod24** [comment]
The failure is because the changelog have trailing whitespace. I'll take a look.

**Comment by Geod24** [comment]
Should be fixed now. Also rebased on master for you and squashed commits.

**Pull request merged**